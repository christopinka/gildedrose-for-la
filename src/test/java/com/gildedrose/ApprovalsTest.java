package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class ApprovalsTest {

    @Test
    public void shouldTestAllCombinations() {


        Integer[] qualityValues = {0, 2, 49, 52};
        Integer[] sellInValues = {-1, 0, 11, 5, 6};
        CombinationApprovals.verifyAllCombinations(this::doUpdateQuality,
            new String[]{
                ItemName.STOCK.getValue(),
                ItemName.BRIE.getValue(),
                ItemName.CONCERT.getValue(),
                ItemName.HAND.getValue(),
                ItemName.MAGIC.getValue()},
            sellInValues,
            qualityValues
        );
    }

    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQualityAndSellIn();
        return ItemPrinter(items[0]);
    }

    private static String ItemPrinter(Item item) {
        return "name: " + ItemName.getNameByValue(item.name)
            + " | sell in: " + item.sellIn
            + " | quality: " + item.quality;
    }

}
