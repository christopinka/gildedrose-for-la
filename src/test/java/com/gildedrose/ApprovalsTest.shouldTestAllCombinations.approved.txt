[Stock, -1, 0] => name: STOCK | sell in: -2 | quality: 0 
[Stock, -1, 2] => name: STOCK | sell in: -2 | quality: 0 
[Stock, -1, 49] => name: STOCK | sell in: -2 | quality: 47 
[Stock, -1, 52] => name: STOCK | sell in: -2 | quality: 50 
[Stock, 0, 0] => name: STOCK | sell in: -1 | quality: 0 
[Stock, 0, 2] => name: STOCK | sell in: -1 | quality: 0 
[Stock, 0, 49] => name: STOCK | sell in: -1 | quality: 47 
[Stock, 0, 52] => name: STOCK | sell in: -1 | quality: 50 
[Stock, 11, 0] => name: STOCK | sell in: 10 | quality: 0 
[Stock, 11, 2] => name: STOCK | sell in: 10 | quality: 1 
[Stock, 11, 49] => name: STOCK | sell in: 10 | quality: 48 
[Stock, 11, 52] => name: STOCK | sell in: 10 | quality: 51 
[Stock, 5, 0] => name: STOCK | sell in: 4 | quality: 0 
[Stock, 5, 2] => name: STOCK | sell in: 4 | quality: 1 
[Stock, 5, 49] => name: STOCK | sell in: 4 | quality: 48 
[Stock, 5, 52] => name: STOCK | sell in: 4 | quality: 51 
[Stock, 6, 0] => name: STOCK | sell in: 5 | quality: 0 
[Stock, 6, 2] => name: STOCK | sell in: 5 | quality: 1 
[Stock, 6, 49] => name: STOCK | sell in: 5 | quality: 48 
[Stock, 6, 52] => name: STOCK | sell in: 5 | quality: 51 
[Aged Brie, -1, 0] => name: BRIE | sell in: -2 | quality: 2 
[Aged Brie, -1, 2] => name: BRIE | sell in: -2 | quality: 4 
[Aged Brie, -1, 49] => name: BRIE | sell in: -2 | quality: 50 
[Aged Brie, -1, 52] => name: BRIE | sell in: -2 | quality: 52 
[Aged Brie, 0, 0] => name: BRIE | sell in: -1 | quality: 2 
[Aged Brie, 0, 2] => name: BRIE | sell in: -1 | quality: 4 
[Aged Brie, 0, 49] => name: BRIE | sell in: -1 | quality: 50 
[Aged Brie, 0, 52] => name: BRIE | sell in: -1 | quality: 52 
[Aged Brie, 11, 0] => name: BRIE | sell in: 10 | quality: 1 
[Aged Brie, 11, 2] => name: BRIE | sell in: 10 | quality: 3 
[Aged Brie, 11, 49] => name: BRIE | sell in: 10 | quality: 50 
[Aged Brie, 11, 52] => name: BRIE | sell in: 10 | quality: 52 
[Aged Brie, 5, 0] => name: BRIE | sell in: 4 | quality: 1 
[Aged Brie, 5, 2] => name: BRIE | sell in: 4 | quality: 3 
[Aged Brie, 5, 49] => name: BRIE | sell in: 4 | quality: 50 
[Aged Brie, 5, 52] => name: BRIE | sell in: 4 | quality: 52 
[Aged Brie, 6, 0] => name: BRIE | sell in: 5 | quality: 1 
[Aged Brie, 6, 2] => name: BRIE | sell in: 5 | quality: 3 
[Aged Brie, 6, 49] => name: BRIE | sell in: 5 | quality: 50 
[Aged Brie, 6, 52] => name: BRIE | sell in: 5 | quality: 52 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => name: CONCERT | sell in: -2 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 2] => name: CONCERT | sell in: -2 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => name: CONCERT | sell in: -2 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 52] => name: CONCERT | sell in: -2 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => name: CONCERT | sell in: -1 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => name: CONCERT | sell in: -1 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => name: CONCERT | sell in: -1 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 52] => name: CONCERT | sell in: -1 | quality: 0 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => name: CONCERT | sell in: 10 | quality: 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 2] => name: CONCERT | sell in: 10 | quality: 3 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => name: CONCERT | sell in: 10 | quality: 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 52] => name: CONCERT | sell in: 10 | quality: 52 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => name: CONCERT | sell in: 4 | quality: 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 2] => name: CONCERT | sell in: 4 | quality: 5 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => name: CONCERT | sell in: 4 | quality: 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 52] => name: CONCERT | sell in: 4 | quality: 52 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => name: CONCERT | sell in: 5 | quality: 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 2] => name: CONCERT | sell in: 5 | quality: 4 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => name: CONCERT | sell in: 5 | quality: 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 52] => name: CONCERT | sell in: 5 | quality: 52 
[Sulfuras, Hand of Ragnaros, -1, 0] => name: HAND | sell in: -1 | quality: 0 
[Sulfuras, Hand of Ragnaros, -1, 2] => name: HAND | sell in: -1 | quality: 2 
[Sulfuras, Hand of Ragnaros, -1, 49] => name: HAND | sell in: -1 | quality: 49 
[Sulfuras, Hand of Ragnaros, -1, 52] => name: HAND | sell in: -1 | quality: 52 
[Sulfuras, Hand of Ragnaros, 0, 0] => name: HAND | sell in: 0 | quality: 0 
[Sulfuras, Hand of Ragnaros, 0, 2] => name: HAND | sell in: 0 | quality: 2 
[Sulfuras, Hand of Ragnaros, 0, 49] => name: HAND | sell in: 0 | quality: 49 
[Sulfuras, Hand of Ragnaros, 0, 52] => name: HAND | sell in: 0 | quality: 52 
[Sulfuras, Hand of Ragnaros, 11, 0] => name: HAND | sell in: 11 | quality: 0 
[Sulfuras, Hand of Ragnaros, 11, 2] => name: HAND | sell in: 11 | quality: 2 
[Sulfuras, Hand of Ragnaros, 11, 49] => name: HAND | sell in: 11 | quality: 49 
[Sulfuras, Hand of Ragnaros, 11, 52] => name: HAND | sell in: 11 | quality: 52 
[Sulfuras, Hand of Ragnaros, 5, 0] => name: HAND | sell in: 5 | quality: 0 
[Sulfuras, Hand of Ragnaros, 5, 2] => name: HAND | sell in: 5 | quality: 2 
[Sulfuras, Hand of Ragnaros, 5, 49] => name: HAND | sell in: 5 | quality: 49 
[Sulfuras, Hand of Ragnaros, 5, 52] => name: HAND | sell in: 5 | quality: 52 
[Sulfuras, Hand of Ragnaros, 6, 0] => name: HAND | sell in: 6 | quality: 0 
[Sulfuras, Hand of Ragnaros, 6, 2] => name: HAND | sell in: 6 | quality: 2 
[Sulfuras, Hand of Ragnaros, 6, 49] => name: HAND | sell in: 6 | quality: 49 
[Sulfuras, Hand of Ragnaros, 6, 52] => name: HAND | sell in: 6 | quality: 52 
[Conjured, -1, 0] => name: MAGIC | sell in: -2 | quality: 0 
[Conjured, -1, 2] => name: MAGIC | sell in: -2 | quality: 0 
[Conjured, -1, 49] => name: MAGIC | sell in: -2 | quality: 45 
[Conjured, -1, 52] => name: MAGIC | sell in: -2 | quality: 48 
[Conjured, 0, 0] => name: MAGIC | sell in: -1 | quality: 0 
[Conjured, 0, 2] => name: MAGIC | sell in: -1 | quality: 0 
[Conjured, 0, 49] => name: MAGIC | sell in: -1 | quality: 45 
[Conjured, 0, 52] => name: MAGIC | sell in: -1 | quality: 48 
[Conjured, 11, 0] => name: MAGIC | sell in: 10 | quality: 0 
[Conjured, 11, 2] => name: MAGIC | sell in: 10 | quality: 0 
[Conjured, 11, 49] => name: MAGIC | sell in: 10 | quality: 47 
[Conjured, 11, 52] => name: MAGIC | sell in: 10 | quality: 50 
[Conjured, 5, 0] => name: MAGIC | sell in: 4 | quality: 0 
[Conjured, 5, 2] => name: MAGIC | sell in: 4 | quality: 0 
[Conjured, 5, 49] => name: MAGIC | sell in: 4 | quality: 47 
[Conjured, 5, 52] => name: MAGIC | sell in: 4 | quality: 50 
[Conjured, 6, 0] => name: MAGIC | sell in: 5 | quality: 0 
[Conjured, 6, 2] => name: MAGIC | sell in: 5 | quality: 0 
[Conjured, 6, 49] => name: MAGIC | sell in: 5 | quality: 47 
[Conjured, 6, 52] => name: MAGIC | sell in: 5 | quality: 50 
